tests/src/test/scala/com/nvidia/spark/rapids/JoinsSuite.scala

```diff
@@ -102,6 +103,21 @@ class JoinsSuite extends SparkQueryCompareTestSuite {
     (A, B) => A.join(B, A("longs") === B("longs"), "LeftAnti")
   }
 
+  IGNORE_ORDER_testSparkResultsAreEqual2(
+    "join condition pushing down for AST non-supported case in outer join",
+    stringWithTailingSpaces, stringWithTailingSpaces2, conf = new SparkConf()) {
+    (A, B) =>
+      A.join(
+        B, length(lower(trim(A("name")))) < length(lower(trim(B("name")))), "leftouter")
+  }
+
+  IGNORE_ORDER_testSparkResultsAreEqual2(
+    "single side join condition pushing down for AST non-supported case in outer join",
+    stringWithTailingSpaces, stringWithTailingSpaces2, conf = new SparkConf()) {
+    (A, B) =>
+      A.join(B, length(lower(trim(A("name")))) < B("number_int"), "leftouter")
+  }
+
```

Comment:
Removed it since duplicated with [join_test.py one](https://github.com/NVIDIA/spark-rapids/pull/9702/files#diff-36070c90652877204943a4cbe235577d7464060469d87bf521dd65fe014f3413R400).